cpp/src/ray/config_internal.cc

```diff
@@ -114,11 +116,11 @@ void ConfigInternal::Init(RayConfig &config, int argc, char **argv) {
   }
   if (worker_type == WorkerType::DRIVER && run_mode == RunMode::CLUSTER) {
     if (redis_ip.empty()) {
-      auto ray_address_env = std::getenv("RAY_ADDRESS");
+      const auto ray_address_env = GetEnvironment("RAY_ADDRESS");
```

Comment:
How about let's just put it into:
https://github.com/ray-project/ray/blob/master/src/ray/common/ray_config_def.h

and use instance() for this?

Reply:
+1

Reply:
I think this should work as a pattern for all the vars. We can recommend to never use raw env vars directly, it has to go through ray_config_def.h

Reply:
I feel ray_config_def.h contains different types of configuration options, which allow users to browse and see what they can change, via env, flags or config files. Do ADDRESS, JOB_ID and RAYLET_PID really belong there? They are not intended to be set by anyone other than Ray. Maybe we can come up with a standard on propagating settings in future.

Reply:
Well, there's the machinery vs the name (e.g., RAY_CONFIG). If you feel strongly about it, how about defining a new macro like RAY_PROCESS_FLAG, which can use the same implementation as RAY_CONFIG under the hood? Then we can define, e.g.:

RAY_PROCESS_FLAG(std::string, JOB_ID, "")

In a separate file for organizational purposes, but still use the same parsing and init logic.

Reply:
If we go this way, I'd suggest `RAY_INTERNAL_FLAG`

Reply:
Both `RAY_INTERNAL_FLAG` and `RAY_PROCESS_FLAG` sound ok to me, maybe `RAY_INTERNAL_FLAG` / `RAY_INTERNAL_CONFIG` is slightly better. Let me know which is the most suitable and I will update the naming later.